File: solution_zzj/278._First_Bad_Version.cc
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        uint l = 1, r = n - 1;
        while (l <= r) {
            uint mid = (l + r) / 2;
            if (isBadVersion(mid))
                r = mid - 1;
            else
                l = mid + 1;
        }
        return l;
    }
};

File: Weijia/278. First Bad Version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution(object):
    def firstBadVersion(self, n):
        s1=0
        s2=n
        while s1<s2:
            mid=(s1+s2)/2
            b=isBadVersion(mid)
            if b==False:
                s1=mid+1
            else:
                s2=mid
        return s1
        
